### PR TITLE
Minor: Convert `Count`'s name to lowercase

### PR DESCRIPTION
**Pull request by jayzhan211**
## Which issue does this PR close?

Part of #10695

### CODE REVIEW

**Review by alamb — APPROVED** [review]
Thank you @jayzhan211 

I merged up from main with this branch to resolve a conflict

File: datafusion/core/src/dataframe/mod.rs [hunk 1]
```diff
@@ -1986,7 +1986,7 @@ mod tests {
 
         assert_batches_sorted_eq!(
             ["+----+-----------------------------+-----------------------------+-----------------------------+-----------------------------+-------------------------------+----------------------------------------+",
-                "| c1 | MIN(aggregate_test_100.c12) | MAX(aggregate_test_100.c12) | AVG(aggregate_test_100.c12) | sum(aggregate_test_100.c12) | COUNT(aggregate_test_100.c12) | COUNT(DISTINCT aggregate_test_100.c12) |",
+                "| c1 | MIN(aggregate_test_100.c12) | MAX(aggregate_test_100.c12) | AVG(aggregate_test_100.c12) | sum(aggregate_test_100.c12) | count(aggregate_test_100.c12) | count(DISTINCT aggregate_test_100.c12) |",
```
Comment (comphead) [inline datafusion/core/src/dataframe/mod.rs:1989]: @jayzhan211 are we planning to align also MIN, MAX, AVG to be lowercase?

**Review by comphead — COMMENTED** [review]
Thanks @jayzhan211 I personally prefer function names to be uppercase, its more readable imho, but what is more important all of it has to be the same case to avoid a mess. Are you planning to convert all others to lowercase as well?

**Comment by alamb** [comment]
> Thanks @jayzhan211 I personally prefer function names to be uppercase, its more readable imho, but what is more important all of it has to be the same case to avoid a mess. Are you planning to convert all others to lowercase as well?

I think the case is being converted as part of the migration to UDAF -- for example https://github.com/apache/datafusion/pull/10964

File: datafusion/core/src/dataframe/mod.rs (hunk 1 above)
Comment (jayzhan211) [reply datafusion/core/src/dataframe/mod.rs:1989]: Yes, we should rename all of them

**Comment by jayzhan211** [comment]
> Thanks @jayzhan211 I personally prefer function names to be uppercase, its more readable imho, but what is more important all of it has to be the same case to avoid a mess. Are you planning to convert all others to lowercase as well?

We can introduce config setting to have upper case for display, but lowercase for internal use (we get the function or compare with it's name)

**Comment by jayzhan211** [comment]
Thanks @alamb and @comphead

**Pull request merged**